arangod/Aql/RegexCache.cpp

```diff
@@ -46,6 +51,40 @@ icu::RegexMatcher* RegexCache::buildLikeMatcher(char const* ptr, size_t length,
 
   return fromCache(_temp, _likeCache);
 }
+
+icu::RegexMatcher* RegexCache::buildSplitMatcher(AqlValue splitExpression, transaction::Methods* trx, bool& isEmptyExpression) {
+
+  std::string rx;
+  
+  AqlValueMaterializer materializer(trx);
+  VPackSlice slice = materializer.slice(splitExpression, false);
+  if (splitExpression.isArray()) {
+    std::string copy;
+    for (auto const& it : VPackArrayIterator(slice)) {
+      copy = it.copyString();
+      if (copy.length() == 0) {
+        // one empty string rules them all
+        isEmptyExpression = true;
+        rx = "";
+        break;
+      }
+      if (rx.size() != 0) {
+        rx += '|';
+      }
+      rx += copy;
```

Comment:
shouldn't `copy` be properly escaped for regex usage here?

Reply:
used https://gist.github.com/edhemphill/1731633  for testing.